### ISSUE TITLE
crash when changing a video option and video driver in the same session

### ISSUE THREAD

**Issue by commiebits** [libretro/RetroArch#4560]
## Description

Retroarch crashes out as soon as you alter a video option when the video driver has been altered to a value that is not the current video driver.

### Expected behavior

Either apply the video driver change right away or defer the video option change till the next session.

### Actual behavior

Crash

### Steps to reproduce the bug

1. With a fresh retroarch, config->driver->video driver->vulkan (from ogl)
2. config->video->bilinear filtering->off (from on)

### Environment information

- OS: Win7

**Comment by JuanVCS** [comment]
Just tried on Arch Linux x86-64 and it also crashes with the steps detailed by op.

**Comment by twinaphex** [comment]
Yo need to restart RetroArch first. I guess we could mention this as a notification message. You can't switch video drivers on the fly right now and assume you can then toggle other settings which require a video driver reinit.

**Comment by commiebits** [comment]
Just a notification won't prevent an accidental change and then crash however. If the driver switch can't be dynamic, in additional to the notification, video config changes should be deferred or disabled to prevent data loss.

**Comment by hizzlekizzle** [comment]
Would it make sense to force a restart after existing the driver menu if any of the drivers change?

**Comment by commiebits** [comment]
Yeah that's reasonable. Many PC games have a yes/no restart prompt whenever they need to alter video settings.
